src/status_im/contexts/wallet/collectible/tabs/overview/view.cljs

```diff
@@ -63,7 +63,9 @@
 
 (defn view
   [collectible]
-  (let [owner-account (rf/sub [:wallet/collectible-details-owner collectible])
+  (let [owner-address (or (rf/sub [:wallet/current-viewing-account-address])
+                          (-> collectible :ownership first :address))
+        owner-account (rf/sub [:wallet-connect/account-details-by-address owner-address])
```

Comment:
I think all of this logic could be wrapped in a single subscription.

@vkjr could you please elaborate on which cases the `current-viewing-account` is `nil` so that we rely on `collectible`'s data? I'm also wondering why we don't always rely on `collectible`'s data

Reply:
The problem with collectible data is that it has information about all the owners inside of it. And when we are interested in balance for particular owner, we need to extract it knowing owner address we are interested in.
If we store collectible as copies with different ownership records that make it harder to display them as a single item on main wallet page (I tried this approach).

`current-viewing-account` can be `nil` on main wallet page but this is totally okay. In this case we calculate collectible balance by summing balances on all accounts we own.